osquery/sql/sqlite_version.cpp

```diff
@@ -0,0 +1,339 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+#include <sqlite3.h>
+
+namespace osquery {
+/**
+ * @brief Compares the position of epoch delimiter between two versions.
+ * Return difference of epoch segment length if both versions have epoch.
+ * Return 1 if left version has epoch but not the right.
+ * Return -1 if right version has epoch but not the left.
+ * Return 0 if epoch doesn't exist in either version.
+ */
+static int compareEpoch(const std::string_view& l_ver,
```

Comment:
nit: It's idiomatic and slightly more performant to pass `string_view` by copy (prevents an additional pointer indirection to the data, prevents aliasing).

Reply:
I've updated the compare functions to pass those by value instead of reference.